Exempt AugAssign from raising reassigning errors

saniQQ opened the pull request:
When applied to a Numpy array, AugAssign does not reallocate the array; hence we should not raise an error.

EmilyBourne reviewed (changes requested):
Could you add a test please. One should be enough :+1:

saniQQ commented:
> Could you add a test please. One should be enough 👍

i added the tests in the Codegen tests is it enough?
i added them there because C still can't handle the augassign.

EmilyBourne commented:
> > Could you add a test please. One should be enough +1
> 
> i added the tests in the Codegen tests is it enough?
> i added them there because C still can't handle the augassign.

I always prefer an end-to-end test. Fortran can handle the augassign and this problem is in the semantic stage so it shouldn't matter which language it's tested with
